src/util.rs

```diff
@@ -1,24 +1,22 @@
 use rand::distributions::Alphanumeric;
-use rand::{self, Rng};
+use rand::{Rng, SeedableRng};
+use rand::rngs::SmallRng;
 use std::ffi::{OsStr, OsString};
 use std::path::{Path, PathBuf};
-use std::{io, str};
+use std::io;
 
 use crate::error::IoResultExt;
 
 fn tmpname(prefix: &OsStr, suffix: &OsStr, rand_len: usize) -> OsString {
     let mut buf = OsString::with_capacity(prefix.len() + suffix.len() + rand_len);
     buf.push(prefix);
 
-    // Push each character in one-by-one. Unfortunately, this is the only
-    // safe(ish) simple way to do this without allocating a temporary
-    // String/Vec.
-    unsafe {
-        rand::thread_rng()
-            .sample_iter(&Alphanumeric)
-            .take(rand_len)
-            .for_each(|b| buf.push(str::from_utf8_unchecked(&[b as u8])))
-    }
+    let small_rng = SmallRng::from_entropy();
+    buf.push(small_rng
+        .sample_iter(&Alphanumeric)
+        .take(rand_len)
+        .collect::<String>()
```

Comment:
```suggestion
        .sample_iter(Alphanumeric)
        .take(rand_len)
        .map(char::from)
        .collect::<String>()
```

Reply:
That's still going to allocate, unfortunately.

Reply:
I'll look for a way to avoid the allocation.